### PR TITLE
[Keyboard] add willoucom/keypad

### PR DESCRIPTION
**Pull request by willoucom**
## Description

Adding a new keyboard  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/willoucom/keypad/info.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            ["GP5", "GP6", "GP7", "GP8", "GP9"],
+            ["GP10", "GP11", "GP12", "GP13", "GP14"],
+            ["GP15", "GP16", "GP17", "GP18", "GP19"],
+            ["GP20"],
+            ["GP21"],
+            ["GP22", "GP26"],
+            ["GP27"],
+            ["GP28"]
+        ],
+        "rows": ["D1"]
```
Comment (dunk2k) [inline keyboards/willoucom/keypad/info.json:28]: 1. Keyboard name adjusted to match directory structure
2. "C2" and "D1" not pins for rp2040 controller
```suggestion
    "keyboard_name": "keypad",
    "maintainer": "willoucom",
    "bootloader": "rp2040",
    "diode_direction": "COL2ROW",
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
    },
    "matrix_pins": {
        "direct": [
            ["GP0", "GP1", "GP2", "GP3", "GP4"],
            ["GP5", "GP6", "GP7", "GP8", "GP9"],
            ["GP10", "GP11", "GP12", "GP13", "GP14"],
            ["GP15", "GP16", "GP17", "GP18", "GP19"],
            ["GP20"],
            ["GP21"],
            ["GP22", "GP26"],
            ["GP27"],
            ["GP28"]
        ]
```

File: keyboards/willoucom/keypad/keymaps/via/keymap.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+    "keyboard": "willoucom/keypad",
```
Comment (dunk2k) [inline keyboards/willoucom/keypad/keymaps/via/keymap.json:2]: move VIA_ENABLE to `keymap.json`
```suggestion
    "keyboard": "willoucom/keypad",
    "config": {
        "features": {
            "via": true
        }
    },
```

File: keyboards/willoucom/keypad/keymaps/via/rules.mk
Comment (dunk2k) [inline keyboards/willoucom/keypad/keymaps/via/rules.mk:1]: if VIA_ENABLE moved to `keymap.json` suggestion accepted, this file can be removed

File: keyboards/willoucom/keypad/config.h
Comment (dunk2k) [inline keyboards/willoucom/keypad/config.h:1]: file can be removed as it doesn't provided or define functionality

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by willoucom** [comment]
Many thanks @dunk2k

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/willoucom/keypad/info.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        "direct": [
+            ["GP0", "GP1", "GP2", "GP3", "GP4"],
+            ["GP5", "GP6", "GP7", "GP8", "GP9"],
+            ["GP10", "GP11", "GP12", "GP13", "GP14"],
+            ["GP15", "GP16", "GP17", "GP18", "GP19"],
+            ["GP20"],
+            ["GP21"],
+            ["GP22", "GP26"],
+            ["GP27"],
+            ["GP28"]
```
Comment (fauxpark) [inline keyboards/willoucom/keypad/info.json:25]: Each row should have the same number of columns. Please rearrange this and update your layout accordingly.

File: keyboards/willoucom/keypad/info.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "force_nkro": true,
+        "pid": "0x4B50",
+        "vid": "0x574A"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
```
Comment (fauxpark) [inline keyboards/willoucom/keypad/info.json:40]: Key index is out of order.
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/willoucom/keypad/info.json (hunk 3 above)
Comment (willoucom) [reply keyboards/willoucom/keypad/info.json:25]: Should I fill the rows with null values?

File: keyboards/willoucom/keypad/info.json (hunk 4 above)
Comment (willoucom) [reply keyboards/willoucom/keypad/info.json:40]: IMG_20230814_122623

For keys 20-28, on the product they are positioned vertically on a daughterboard which can be moved and rotated. To make the visual more attractive I made the decision to use the horizontal space and move this section up to make the configurator easier to use.
Should I position this section below the pad grouping the keys 0 to 19?

File: keyboards/willoucom/keypad/info.json (hunk 3 above)
Comment (fauxpark) [reply keyboards/willoucom/keypad/info.json:25]: No, rearrange the pins into something more rectangular.

File: keyboards/willoucom/keypad/info.json (hunk 4 above)
Comment (fauxpark) [reply keyboards/willoucom/keypad/info.json:40]: The issue isn't the physical positioning.

Comment (willoucom) [reply keyboards/willoucom/keypad/info.json:59]: I guess it's the labels that are wrong, I fixed these as follows
image

File: keyboards/willoucom/keypad/info.json (hunk 3 above)
Comment (fauxpark) [reply keyboards/willoucom/keypad/info.json]: If the pin count is prime, then sure.

File: keyboards/willoucom/keypad/info.json (hunk 4 above)
Comment (willoucom) [reply keyboards/willoucom/keypad/info.json:59]: Please can you explain, I don't understand what you mean.

Comment (fauxpark) [reply keyboards/willoucom/keypad/info.json:59]: image

File: keyboards/willoucom/keypad/info.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                "GP16",
+                "GP17",
+                "GP18",
+                "GP19",
+                "GP20",
+                "GP21",
+                "GP22",
+                "GP26",
+                "GP27",
+                "GP28"
```
Comment (fauxpark) [inline keyboards/willoucom/keypad/info.json:43]: 26 is not prime. Try again.

File: keyboards/willoucom/keypad/info.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (fauxpark) [inline keyboards/willoucom/keypad/info.json:1]: Can you please run `qmk format-json` on this file.

File: keyboards/willoucom/keypad/keymaps/default/keymap.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard": "willoucom/keypad",
+    "layout": "LAYOUT",
+    "layers": [
+        [
+            "KC_GRV", "KC_1", "KC_2", "KC_3", "KC_4",           "KC_ESC",
+            "KC_LSFT", "KC_Q", "KC_W", "KC_E", "KC_R",          "KC_LBRC",
+            "KC_LCTL", "KC_A", "KC_S", "KC_D", "KC_F",      "KC_ENT", "KC_G",
+            "KC_SPC", "KC_TAB", "KC_Z", "KC_X", "KC_C",         "KC_RBRC",
+                                                                "KC_M"
```
Comment (fauxpark) [inline keyboards/willoucom/keypad/keymaps/default/keymap.json:10]:
```suggestion
            "KC_GRV",  "KC_1",    "KC_2",    "KC_3",    "KC_4",                "KC_ESC",
            "KC_LSFT", "KC_Q",    "KC_W",    "KC_E",    "KC_R",                "KC_LBRC",
            "KC_LCTL", "KC_A",    "KC_S",    "KC_D",    "KC_F",     "KC_ENT",             "KC_G",
            "KC_SPC",  "KC_TAB",  "KC_Z",    "KC_X",    "KC_C",                "KC_RBRC",
                                                                               "KC_M"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/willoucom/keypad/info.json (hunk 6 above)
Comment (willoucom) [reply keyboards/willoucom/keypad/info.json:1]: done

**Review by fauxpark — APPROVED** [review]

**Comment by willoucom** [comment]
Many thanks @fauxpark

**Pull request merged**